services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/ResumableFileDownload.java

```diff
@@ -125,6 +152,153 @@ public Optional<Long> totalSizeInBytes() {
         return Optional.ofNullable(totalSizeInBytes);
     }
 
+    @Override
+    public String toString() {
+        return ToString.builder("ResumableFileDownload")
+                       .add("bytesTransferred", bytesTransferred)
+                       .add("fileLastModified", fileLastModified)
+                       .add("s3ObjectLastModified", s3ObjectLastModified)
+                       .add("totalSizeInBytes", totalSizeInBytes)
+                       .add("downloadFileRequest", downloadFileRequest)
+                       .build();
+    }
+
+    /**
+     * Persists this download object to a file in Base64-encoded JSON format.
+     *
+     * @param path The path to the file to which you want to write the serialized download object.
+     */
+    public void writeToFile(Path path) {
+        try {
+            Files.write(path, toBase64Bytes(ResumableFileDownloadSerializer.toJson(this)));
+        } catch (IOException e) {
+            throw SdkClientException.create("Failed to write to " + path, e);
+        }
+    }
+
+    /**
+     * Writes the serialized JSON data representing this object to an output stream.
+     * Note that the {@link OutputStream} is not closed or flushed after writing.
+     *
+     * @param outputStream The output stream to write the serialized object to.
+     */
+    public void writeToOutputStream(OutputStream outputStream) {
+        byte[] bytes = ResumableFileDownloadSerializer.toJson(this);
+        try {
+            ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(bytes);
+            IoUtils.copy(byteArrayInputStream, outputStream);
+        } catch (IOException e) {
+            throw SdkClientException.create("Failed to write this download object to the given OutputStream", e);
+        }
+    }
+
+    /**
+     * Returns the serialized JSON data representing this object as a UTF-8 string.
+     */
+    public String toUtf8String() {
+        return new String(ResumableFileDownloadSerializer.toJson(this), StandardCharsets.UTF_8);
+    }
+
+    /**
+     * Returns the serialized JSON data representing this object as a string encoded with the supplied charset.
+     *
+     * @param cs encoding charset
+     * @return the serialized JSON string
+     */
+    public String toString(Charset cs) {
+        return new String(ResumableFileDownloadSerializer.toJson(this), cs);
+    }
+
+    /**
+     * Returns the serialized JSON data representing this object as an {@link SdkBytes} object.
+     *
+     * @return the serialized JSON as {@link SdkBytes}
+     */
+    public SdkBytes toBytes() {
+        return SdkBytes.fromByteArray(ResumableFileDownloadSerializer.toJson(this));
+    }
+
+    /**
+     * Returns the serialized JSON data representing this object as a {@link ByteArrayInputStream}.
```

Comment:
nit: IMO `ByteArrayInputStream` is implementation detail and should not be included in the docs

Reply:
good point